### DISCUSSION TITLE
How to improve re-validation cases with items right next to eachother?

### DISCUSSION THREAD

**Discussion by tordans** [streetcomplete/StreetComplete#3700] [Feedback]
In Neukölln, Berlin, we mapped a lot of bike parking amenities. I know from explaining to new users, that one question is, how to handle cases where bike stands are a group but there is some other object in between (tree, street cabinet, street light and such). Our local mapping suggestion is, to add two nodes in those cases and map the other object as well.

This leads to situations, where there are two bike stand nodes right next to eachother.

In SC, those get re-validated. One is a pin, the other a little "future question" circle (referenced as circle below)

The issue is: When being asked about the number of bike stands, a SC user will not know that the circle is also a bike stand. So its a hard decision to count all or only some bike stands when answering the question.

Example: 

![image](https://user-images.githubusercontent.com/111561/151715759-2733b5ea-1356-482c-bbba-baade6ebffe7.png)

I see, that SC has this issue considered with objects that are a bit further away:

![image](https://user-images.githubusercontent.com/111561/151715764-78b8e91f-e115-4ce2-b011-2a5575cc7a96.png)

So I wonder, is there a smart way to improve the first mapping experience in the first screenshot, similar to what the second screenshot already does?

Maybe add a second line below the note, right above the input-section?

> FYI, there is another bike stand right next to this with capacity 4, so only consider this group of stands.

**Comment by matkoniecz** [comment]
SC attempts to display an icons/areas of nearby objects but it apparently fails to show objects extremely close to the quest location.

Is there a valid case of tagging this stands exactly next to each other as a separate objects? Is it in a location where you would be able to take a photo or link Mapillary/GSV/etc ? Can you link OSM data at this location?

In general StreetComplete is unable to handle all kinds of invalid data, but should be able to handle valid one.

**Reply by Helium314** [reply]
In case of wheelbenders next to stands it's definitely valid, so such nearby stands need to be considered anyway.

**Reply by tordans** [reply]
Here are a few examples:

- https://www.openstreetmap.org/edit?node=6268105071#map=20/52.47531/13.44410 with a street lamp in between (the first screenshot above)
- https://www.openstreetmap.org/node/4978245166 with a sidewalk in between and one stand on private property and the other on the sidewalk
- and I assume https://www.openstreetmap.org/way/772848932 would be such a case as well with one area dedicated for cargo and one for regular bikes, both next to one another

There is a ton of Mapillary images available (a lot of 360° as well).

**Reply by matkoniecz** [reply]
> In case of wheelbenders next to stands it's definitely valid, so such nearby stands need to be considered anyway.

That reminds me about case when I run into it and wished that bicycle quest would show me bicycle parking type if tagged already as there was stand next to wheelbender.

It predated showing nearby icons so I dropped this idea but maybe it should be resurrected.

**Reply by peternewman** [reply]
Couldn't it just do the OSM feature stuff to name it based on the preset detected, hopefully less fiddly logic needed and it could also leverage NSI to show names for docking bike systems etc.

**Comment by FloEdelmann** [comment]
See also https://github.com/streetcomplete/StreetComplete/issues/3681.

**Comment by westnordost** [comment]
I don't understand the issue.

> I see, that SC has this issue considered with objects that are a bit further away:

... is wrong. All surrounding bike parkings are shown when selecting a bike parking quest, even those that are just one meter away. Or at least should be. If they are not, this is a bug.

**Reply by peternewman** [reply]
Except when they're under the quest marker @westnordost !

Depending on the orientation of the map when answering https://www.openstreetmap.org/node/6268105071 depends on whether you see none or all of the other one, or only a little bit of the square!

![Screenshot_20220205-022355](https://user-images.githubusercontent.com/6216436/152625363-604c36f4-40c7-4202-b52a-f73e84747ea3.png)
